lib/gollum/app.rb

```diff
@@ -179,7 +179,7 @@ def wiki_new
       halt 500 unless tempfile.is_a? Tempfile
 
       # Remove page file dir prefix from upload path if necessary -- committer handles this itself
-      dir      = wiki.per_page_uploads ? params[:upload_dest].match(/^(#{wiki.page_file_dir}\/+)?(.*)/)[2] : 'uploads'
+      dir      = wiki.per_page_uploads ? params[:upload_dest] : ::File.join([wiki.page_file_dir, 'uploads'].compact)
```

Comment:
And return a BAD REQUEST if it isn't? I think it's not a good idea, because pages can be not under `page_file_dir`. The same should go for files. I understand, that this is risky, because someone can actually overwrite a project file, but enabling uploads is for internal use only anyway (i guess, or rather hope).